src/main/java/org/mockito/internal/matchers/InstanceOf.java

```diff
@@ -30,6 +30,11 @@ public boolean matches(Object actual) {
                         || clazz.isAssignableFrom(actual.getClass()));
     }
 
+    @Override
+    public Class<?> type() {
+        return clazz;
+    }
```

Comment:
Exposing this provides solution to https://github.com/mockito/mockito/issues/1593